core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2418,6 +2418,21 @@ private[spark] object Utils extends Logging {
       sparkJars.map(_.split(",")).map(_.filter(_.nonEmpty)).toSeq.flatten
     }
   }
+
+  def setCallerContext(context: String): Boolean = {
+    var succeed = false
+    try {
+      val Builder = Utils.classForName("org.apache.hadoop.ipc.CallerContext$Builder")
+      val builderInst = Builder.getConstructor(classOf[String]).newInstance(context)
+      val ret = Builder.getMethod("build").invoke(builderInst)
+      val callerContext = Utils.classForName("org.apache.hadoop.ipc.CallerContext")
```

Comment:
It might be nice to do this first outside of this try block and it the class isn't found we just skip the rest of this and assume its before hadoop 2.8.0